### DISCUSSION TITLE
Feature Request: Support Vitest

### DISCUSSION THREAD

**Discussion by NateRadebaugh** [help-me-mom/ng-mocks#5064] [Q&A] [answered]
### Describe the feature or problem you'd like to solve

I know jest is supported, but is Vitest? The syntax is very similar.

**Comment by satanTime** [comment]
Hey @NateRadebaugh, 

personally, I don't have experience with `Vitest`, but if it's very similar it should work.
The only thing `ng-mocks` requires is `angular`, `beforeEach`, `afterEach`, `beforeAll` and `afterAll`.

So if `Vitest` is providing them, `ng-mocks` should be compatible with it.

**Comment by Tallyb** [comment]
it needs the equivalent of jest-preset-angular

**Reply by satanTime** [reply]
Hi @Tallyb, do you have an example of a project I could take a look at to understand how `Vitest` works and where `ng-mocks` can fit?

**Reply by Tallyb** [reply]
https://github.com/Tallyb/ng-mocks-vitest @satanTime

**Reply by satanTime** [reply]
Thank you a lot and Merry Christmas! I'll take a look next days.

**Comment by mionelTX** [comment]
Could you please consider adding Vitest support to ng-mocks? It's 2025, and Vitest has become increasingly popular due to its speed, native ESM support, and seamless integration with Vite. With Angular 18 adopting Vite as its default build system, many developers are transitioning to Vitest for testing. Tools like AnalogJS have made integrating Vitest into Angular projects straightforward. Adding support for Vitest in ng-mocks would greatly benefit the community.

**Comment by Tallyb** [comment]
I have only checked services and not rendering (we are using storybook for testing UI), and it works just fine. It is basically using the analogjs plugin for compiling angular. 
Feel free to consult this repo https://github.com/Tallyb/angular-realworld-example-app as reference.

**Comment by ytchenak** [comment]
I noted that angular 20 added experimental vitest browser support to unit-testing: see PR https://github.com/angular/angular-cli/pull/30168

**Comment by undsoft** [comment]
Note, that if you are using MockComponent and you may encounter errors related to missing providers such as `No provider found for x`.

You can fix them by adjusting your `vite.config.ts`:
```
    server: {
      deps: {
        inline: ['ng-mocks'],
      },
    },
```

**Comment by GipHub123** [comment]
I tried running my test cases with Vitest, but I had previously run them successfully using Karma.

```
Test Files  302 failed | 1942 passed (2244)
   Tests  494 failed | 7757 passed (8251)
   Errors  92 errors
   Start at  14:46:27
   Duration  1640.51s (transform 392.43s, setup 94.65s, collect 3782.75s, tests 980.44s, environment 10146.52s, prepare 908ms)
```
   
Log was filled with weird error like. 
```
Failed to construct 'PointerEvent': member view is not of type Window
Error: Component '_XXXComponent' has unresolved metadata. Please call `await TestBed.compileComponents()` before running this test.
Error: [vitest-worker]: Closing rpc while "fetch" was pending
```

Vitest seems quite slow—it took around 27 minutes, whereas the same tests ran in about 6-7 minutes with Karma.
Note that I'm not sure if problems are ng-mocks / vitest / jsdom or test case related 😐

**Comment by satanTime** [comment, marked as answer]
Update: native Angular Vitest support is now part of ng-mocks and was released in [v14.17.0](https://github.com/help-me-mom/ng-mocks/releases/tag/v14.17.0).

The tested matrix covers Angular 20 with Vitest 3 and jsdom 26 in zoneless mode, and Angular 21–22 with Vitest 4 and jsdom 28 in both zoned and zoneless modes. The required configuration is documented in the [Angular native Vitest setup](https://ng-mocks.sudo.eu/extra/install#angular-native-vitest-setup).

Closing as implemented.
